### PR TITLE
gh-124182: Explain naming rules for struct sequence types

### PR DESCRIPTION
**Pull request by ffelixg**
* Issue: gh-124182

----
📚 Documentation preview 📚: https://cpython-previews--124335.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/tuple.rst [hunk 1]
```diff
@@ -167,7 +167,13 @@ type.
 
    .. c:member:: const char *name
 
-      Name of the struct sequence type.
+      Pointer to a NUL-terminated string containing the fully qualified name of the
+      type.  For types that are accessible as module globals, the string should be
+      the full module name, followed by a dot, followed by the type name.  If the
+      module is a submodule of a package, the full package name is part of the full
```
Comment (vstinner) [inline Doc/c-api/tuple.rst]: "If the module is a submodule (...)" IMO that's too much, you can remove it.

File: Doc/c-api/tuple.rst [hunk 2]
```diff
@@ -167,7 +167,13 @@ type.
 
    .. c:member:: const char *name
 
-      Name of the struct sequence type.
+      Pointer to a NUL-terminated string containing the fully qualified name of the
+      type.  For types that are accessible as module globals, the string should be
```
Comment (vstinner) [inline Doc/c-api/tuple.rst]: I suggest to replace the whole paragraph with just:

"Fully qualified name of the type; null-terminated UTF-8 encoded."

Comment (ffelixg) [reply Doc/c-api/tuple.rst]: Fine with me. I think it would have been enough to put me on the right track when I ran into the issue.

**Comment by vstinner** [comment]
@encukou: Would you mind to review this change? Does it explain correctly that the name must contain a module name?

**Comment by JelleZijlstra** [comment]
It might be worth being more explicit and saying the name must include the module name, and given an example like `os.terminal_size`. (Aside: several structseqs in posixmodule.c do not include the module name. Should they be fixed?)

File: Doc/c-api/tuple.rst [hunk 3]
```diff
@@ -167,7 +167,7 @@ type.
 
    .. c:member:: const char *name
 
-      Name of the struct sequence type.
+      Fully qualified name of the type; null-terminated UTF-8 encoded.
```
Comment (vstinner) [inline Doc/c-api/tuple.rst:170]:
```suggestion
      Fully qualified name of the type; null-terminated UTF-8 encoded.
      The name must contain the module name.
```

**Comment by ffelixg** [comment]
> It might be worth being more explicit and saying the name must include the module name, and given an example like `os.terminal_size`. (Aside: several structseqs in posixmodule.c do not include the module name. Should they be fixed?)

Looks to me like all of the incorrect names in posixmodule.c get overwritten to include the module name before they are accessed. Not sure why they didn't get the right name from the beginning, seems to work just as well.

**Review by vstinner — APPROVED** [review]
LGTM.

@JelleZijlstra: Do you want to double check the change?

**Review by JelleZijlstra — APPROVED** [review]

**Comment by vstinner** [comment]
Merged, thanks @ffelixg for the doc and @JelleZijlstra for the reviews ;-)

**Pull request merged**